packages/cli/generation/ir-generator/src/converters/services/convertHttpRequestBody.ts

```diff
@@ -43,13 +44,27 @@ export function convertHttpRequestBody({
             name: file.casingsGenerator.generateName(fileUploadRequest.name),
             properties: fileUploadRequest.properties.map((property) => {
                 if (property.isFile) {
-                    return FileUploadRequestProperty.file({
-                        key: file.casingsGenerator.generateNameAndWireValue({
-                            wireValue: property.key,
-                            name: property.key
-                        }),
-                        isOptional: property.isOptional
-                    });
+                    if (property.isArray) {
+                        return FileUploadRequestProperty.file(
+                            FileProperty.fileArray({
+                                key: file.casingsGenerator.generateNameAndWireValue({
+                                    wireValue: property.key,
+                                    name: property.key
+                                }),
+                                isOptional: property.isOptional
```

Comment:
I don't think we'll support `list<optional<file>>`, thoughts on disallowing that ?

Reply:
`optional<list<file>>` would make sense though

Reply:
This is for optional<list<file>>, not list<optional<file>>

Reply:
It tries to follow the existing paradigm which is isOptional flags theproperty as optional:
<img width="372" alt="image" src="https://github.com/fern-api/fern/assets/11140328/c42f59de-7092-45c7-84a3-cf556f56ade8">

Reply:
You can also see where this is consumed, it makes `optional<list<file>>`, not `list<optional<file>>`

Reply:
ah thank you!